src/memory.c

```diff
@@ -26,7 +26,7 @@ size_t copy_address(phandle pid, void *addr, void *data, size_t len) {
   remote[0].iov_base = addr;
   remote[0].iov_len = len;
 
-  size_t bytes = process_vm_readv(pid, local, 1, remote, 1, 0);
+  ssize_t bytes = process_vm_readv(pid, local, 1, remote, 1, 0);
   if (bytes < 0) {
```

Comment:
As previously written I think this condition will never happen, as values less than 0 cannot be stored in a `size_t`

It is not clear if we should also change the signature of copy_address to return a `ssize_t`, are you expecting this to return -1 on error? If so that would also need to change to `ssize_t`.

Reply:
Oh, excellent catch. Of course they should all be `ssize_t`, yes, since we use -1 as the return value.

Reply:
The `ReadProcessMemory()` API in Windows does use `size_t` and could in theory overflow `ssize_t`, but it's irrelevant to the size of slices of memory being read by this program.